client/profile-wizard/steps/store-details/index.js

```diff
@@ -198,7 +200,10 @@ class StoreDetails extends Component {
 	}
 
 	validateStoreDetails( values ) {
-		const errors = validateStoreAddress( values );
+		const { getLocale } = this.props;
+		const locale = getLocale( values.countryState );
+		const validateAddress = getStoreAddressValidator( locale );
```

Comment:
I'm really not a fan of this and wish we could handle this in the child component so we're not dealing with the locale data store in the parent and child.  But I'm a little stumped on a pattern that would work here.

We could add an `onValidate` hook from the form to filter errors again, but this feels messy.

```jsx
<Form
  validate={ parentValidation }
>
  <StoreAddress
    // onValidate?
  />
</Form>
```

Reply:
Yeah this does seem like the easiest approach atm, and I am not a big fan of the `onValidate` hook as well.
I think another approach would be to refactor things a bit, and move the email address field in it's own component, we could then pass a `setError` state or something like that, I guess similar to `onValidate` that sets a local error state. But not sure if this will cause some discrepancies for when `onValidate` is called by the Form component and when a field has updated the validation errors.

Your current approach might be ok for now.